tests/processing/strategies/test_batching.py

```diff
@@ -1,17 +1,48 @@
 import time
 from datetime import datetime
-from typing import Any, MutableSequence, Sequence
+from typing import Any, Callable, Mapping, MutableSequence, Sequence, TypeVar
 from unittest.mock import patch
 
 from arroyo.backends.local.backend import LocalBroker as Broker
 from arroyo.backends.local.backend import LocalConsumer
 from arroyo.commit import IMMEDIATE
 from arroyo.processing.processor import StreamProcessor
+from arroyo.processing.strategies.abstract import (
+    ProcessingStrategy,
+    ProcessingStrategyFactory,
+)
 from arroyo.processing.strategies.batching import (
     AbstractBatchWorker,
-    BatchProcessingStrategyFactory,
+    BatchProcessingStrategy,
 )
-from arroyo.types import Message, Topic
+from arroyo.types import Message, Partition, Position, Topic
+
+TPayload = TypeVar("TPayload")
+TResult = TypeVar("TResult")
+
+
+class BatchProcessingStrategyFactory(ProcessingStrategyFactory[TPayload]):
```

Comment:
Why is this still needed in the tests?

Reply:
Mostly I was lazy to refactor the tests since it will be deleted soon. But I think i will make the change now as it's not really needed.

Reply:
Rewrote the tests so it can be removed